src/WorkOS.net/Services/UserManagement/Entities/Reasons.cs

```diff
@@ -0,0 +1,24 @@
+namespace WorkOS
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Reasons for unauthorized access to an organization the user is a member of.
+    /// </summary>
+    public class Reasons
+    {
+        /// <summary>
+        /// The type of reason for unauthorized access to an organization.
+        /// </summary>
+        [JsonProperty("type")]
+        public Type Type { get; set; }
+
+        /// <summary>
+        /// Array of allowed authentication methods allowed by the organziation.
+        /// </summary>
+        [JsonProperty("allowed_authentication_methods")]
+        public AuthenticationMethods[] AllowedAuthenticationMethods { get; set; }
```

Comment:
Using a `List` here would be more idiomatic:

```suggestion
        public List<AuthenticationMethods> AllowedAuthenticationMethods { get; set; }
```